packages/web3-utils/src/utils.js

```diff
@@ -528,10 +534,11 @@ var sha3Raw = function(value) {
  *
  * @method toNumber
  * @param {String|Number|BN} value
- * @return {Number}
+ * @param {Boolean} bigIntOnOverflow - if true, return the hex value in case of overflow
+ * @return {Number|String}
  */
-var toNumber = function(value) {
-    return typeof value === 'number' ? value : hexToNumber(toHex(value));
+var toNumber = function (value, bigIntOnOverflow) {
```

Comment:
looks like the `bigIntOnOverflow` param should be optional and false by default in all functions.What do you think?

Reply:
yes @avkos it is optional and therefore it defaults to `undefined` in both `toNumber` and hexToNumber`. However, I did not update all other functions because I did not want to invest more time on 1.x.